File: ej4.py
# Lista inicial
lista_inicial = ['rojo', 'verde', 'azul', 'amarillo', 'gris', 'blanco', 'negro']
print(f"lista original: {lista_inicial}")
# Lista de elementos a borrar
elementos_a_borrar = ['amarillo', 'café', 'blanco']
print(f"Elementos a eliminar: {elementos_a_borrar}")
# Eliminar elementos indicados usando lambda
lista_resultante = list(filter(lambda x: x not in elementos_a_borrar, lista_inicial))

# Mostrar el resultado
print(f"Lista resultante: {lista_resultante}")

File: ej4.hs
-- Sitio para la ejecucion del programa https://www.tutorialspoint.com/compile_haskell_online.php


["amarillo"]
["amarillo", "blanco"]
["amarillo", "café", "blanco"]
["rojo", "verde", "azul", "amarillo", "gris", "blanco", "negro"]
["morado", "naranja"]


["fruta", "verde", "azul", "rojo", "amarillo", "verde", "fruta", "rojo", "blanco"]
["fruta", "verde", "blanco"]

-- Lista inicial
listaInicial :: [String]
listaInicial = ["rojo", "verde", "azul", "amarillo", "gris", "blanco", "negro"]

-- Lista de elementos a borrar
elementosABorrar :: [String]
elementosABorrar = ["amarillo", "café", "blanco"]

-- Eliminar elementos indicados usando lambda
listaResultante :: [String]
listaResultante = filter (\x -> not (elem x elementosABorrar)) listaInicial

-- Mostrar el resultado
main :: IO ()
main = print listaResultante
